massa-grpc/src/stream/send_blocks.rs

```diff
@@ -0,0 +1,194 @@
+                                let (rest, res_block): (&[u8], SecureShareBlock) = tuple;
+                                if rest.is_empty() {
+                                    if let Err(e) = res_block
+                                        .verify_signature()
+                                        .and_then(|_| res_block.content.header.verify_signature())
+                                        .map(|_| {
+                                            res_block
+                                                .content
+                                                .header
+                                                .content
+                                                .endorsements
+                                                .iter()
+                                                .map(|endorsement| endorsement.verify_signature())
+                                                .collect::<Vec<Result<(), ModelsError>>>()
+                                        })
+                                    {
+                                        // Signature error
+                                        send_blocks_notify_error(
+                                            req_content.id.clone(),
+                                            tx.clone(),
+                                            tonic::Code::InvalidArgument,
+                                            format!("wrong signature: {}", e),
+                                        )
+                                        .await;
+                                        continue;
+                                    }
+
+                                    let block_id = res_block.id;
+                                    let slot = res_block.content.header.content.slot;
+                                    let mut block_storage = storage.clone_without_refs();
+                                    block_storage.store_block(res_block.clone());
+                                    consensus_controller.register_block(
+                                        block_id,
+                                        slot,
+                                        block_storage.clone(),
+                                        false,
+                                    );
+
+                                    if let Err(e) =
+                                        protocol_sender.integrated_block(block_id, block_storage)
+                                    {
+                                        send_blocks_notify_error(
+                                            req_content.id.clone(),
+                                            tx.clone(),
+                                            tonic::Code::Internal,
+                                            format!("failed to propagate block: {}", e),
+                                        )
+                                        .await;
+                                        // continue ?
```

Comment:
yes, we send back an error message about this block id. May be the next block is valid.